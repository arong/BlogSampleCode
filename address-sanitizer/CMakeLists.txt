cmake_minimum_required(VERSION 3.28)

project(AsanSamples CXX)

add_compile_options(-fsanitize=address -g)
add_link_options(-fsanitize=address)

add_executable(global-buffer-overflow global-buffer-overflow.cpp)
add_executable(init-order init-order.cpp init-order-b.cpp)
add_executable(memory-leak memory-leak.cpp)
add_executable(out-of-bound out-of-bound.cpp)
add_executable(stack-buffer-overflow stack-buffer-overflow.cpp)
add_executable(use-after-free use-after-free.cpp)
add_executable(use-after-return use-after-return.cpp)
add_executable(use-after-scope use-after-scope.cpp)
