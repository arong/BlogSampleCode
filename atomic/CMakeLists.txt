cmake_minimum_required(VERSION 3.28)
project(CPPAtomic CXX)

set(CMAKE_CXX_STANDARD 23)

add_executable(demo1 atomic.cpp)
add_executable(leader compare_exchange_strong.cpp)
add_executable(wait wait_notify.cpp)