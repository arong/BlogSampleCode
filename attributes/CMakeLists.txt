cmake_minimum_required(VERSION 3.28)

project(FilesystemDemo CXX)

set(CMAKE_CXX_STANDARD 23)

add_executable(nodiscard nodiscard.cpp)
add_executable(nodiscard20 nodiscard20.cpp)
add_executable(maybe_unused maybe_unused.cpp)
add_executable(fallthrough fallthrough.cpp)
add_executable(deprecated deprecated.cpp)
add_executable(no_unique_address no_unique_address.cpp)
add_executable(assume assume.cpp)
add_executable(carries_dependency carries_dependency.cpp)
