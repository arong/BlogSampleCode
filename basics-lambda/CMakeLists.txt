cmake_minimum_required(VERSION 3.28)
project(Lambda CXX)
set(CMAKE_CXX_STANDARD 20)


add_executable(e1 e1_sort.cpp)
add_executable(e2 e2_algo.cpp)
add_executable(e3 e3_capture.cpp)
add_executable(e4 e4_high_order.cpp)
add_executable(e5 e5_generic.cpp)
