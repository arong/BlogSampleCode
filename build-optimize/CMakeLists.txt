cmake_minimum_required(VERSION 3.28)

project(OptimizeBuild CXX)

add_executable(demo demo.cpp)
target_precompile_headers(demo INTERFACE pch.h)
