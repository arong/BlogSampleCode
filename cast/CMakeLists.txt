cmake_minimum_required(VERSION 3.28)

project(CastingDemo CXX)

set(CMAKE_CXX_STANDARD 20)

add_compile_options(-Wall)
add_executable(implicit-conversion 1-implicit-conversion.cpp)
add_executable(arithmetic-conversion 2-arithmetic-conversion.cpp)
add_executable(signed-conversion 3-signed-conversion.cpp)
add_executable(user-conversion 4-user-conversion.cpp)

add_executable(static-cast 5-static_cast.cpp)
add_executable(multi-hops 6-multi-hops.cpp)

add_executable(type-aliasing 14-type-aliasing.cpp)
add_executable(inheritance 7-inheritance.cpp)
add_executable(downcasting 8-downcasting.cpp)
add_executable(const_cast 9-const_cast.cpp)
add_executable(member-overload 10-member-overload.cpp)
add_executable(dynamic_cast 11-dynamic_cast.cpp)
add_executable(reinterpret_cast 12-reinterpret_cast.cpp)
add_executable(reinterpret-cast-private 13-reinterpret-cast-private.cpp)
