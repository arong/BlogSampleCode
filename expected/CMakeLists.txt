cmake_minimum_required(VERSION 3.28)
project(ExpectedDemo CXX)

set(CMAKE_CXX_STANDARD 23)

# 基本样例
add_executable(basic expect.cpp)

# transform 样例
add_executable(transform transform.cpp)

# and_then 样例
add_executable(and_then and_then.cpp)

# transform_error 样例
add_executable(transform_error transform_error.cpp)

# or_else 样例
add_executable(or_else or_else.cpp)
